src/HID-APIs/SurfaceDialAPI.hpp

```diff
@@ -81,10 +106,44 @@ void SurfaceDialAPI::release(void)
 void SurfaceDialAPI::releaseAll(void)
 {
 	_button = false;
-	rotate(0);
+	update();
+}
+
+void SurfaceDialAPI::onScreen(bool s)
+{
+	_onScreen = s;
+	update();
+}
+
+void SurfaceDialAPI::update()
+{
+	HID_SurfaceDialReport_Data_t report;
+	report.button = _button;
+	report.rotation = 0;
+	if(_onScreen)
+	{
+		report.xAxis = _xAxis;
```

Comment:
validation of a correct value is not done yet

Reply:
how should the values be validated?

Reply:
ah, I see in the docs that you can specify values out of range to remove the device from the screen

Reply:
Which Docs say that you can specify values out of the range to remove the device from the screen?